docs/CONFIG-VARS.md

```diff
@@ -387,3 +388,101 @@ postgres_servers = {
   }
 }
 ```
+
+## Azure Application Gateway with Azure Web Application Firewall (WAF)
+
+Azure Web Application Firewall (WAF) on Azure Application Gateway provides centralized protection from common exploits and vulnerabilities. Please see the [sample input file](../examples/sample-input-app-gateway.tfvars) for Application Gateway creation example.
+
+| Name | Description | Type | Default | Notes |
+| :--- | :--- | :--- | :--- | :--- |
+| create_app_gateway | Creates Azure Application Gateway | bool | false | |
+| app_gateway_config | Map of Application Gateway configuration objects | map | null | All the variables that can be defined in the `app_gateway_config` are described in table below. |
+| waf_policy | A JSON file with all the WAF_Policy rules | map | null | The WAF policy has few required components see the details below. |
+
+The `app_gateway_config` variable can contain none, some, or all of the parameters listed below:
+For the details of all the parameters that can be specified see: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/application_gateway
+
+| Name | Description | Type | Required | Default | Notes |
+| :--- | :--- | :--- | :--- | :--- | :--- |
+| sku | The Name of the SKU to use for this Application Gateway. | string | false | "Standard_v2" | If WAF policy is enabled the default is `"WAF_v2"` |
```

Comment:
Is there a reason we're not always enabling WAF_v2 and adding in the base WAF policy provided by the application gateway? Will folks want a Standard_v2 gateway and then want to add a WAF? If they do I don't believe that will work.

Reply:
WAF_v2 sku can only be enabled with WAF and during my testing I haven't come across any recommended base WAF policy provided by app gateway. Which makes this variable dependent on user input. So if user does not add this variable, app gateway creation will continue without WAF with Standard_v2 sku.
If there is a base WAF policy that we should add as default then we can definitely change this logic and have WAF_v2 as the default.